projects/demo/src/app/forms/dynamic-controls/dynamic-controls.demo.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2016-2023 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Component, inject } from '@angular/core';
+import { FormControl, UntypedFormBuilder, UntypedFormGroup, Validators } from '@angular/forms';
+
+@Component({
+  selector: 'dynamic-controls-demo',
+  templateUrl: './dynamic-controls.demo.html',
+})
+export class DynamicControlsDemo {
+  // radio control
+  readonly certValidation = new FormControl<string | null>(null, Validators.required);
+
+  fb = inject(UntypedFormBuilder);
+
+  form: UntypedFormGroup = this.fb.group({
+    platform: new FormControl<string | null>(null, [Validators.required]),
+  });
+
+  ngOnInit() {
+    this.registerPlatformChange();
+
+    this.registerCertValidateChange();
+  }
+
+  private registerPlatformChange() {
+    this.form.controls.platform.valueChanges.subscribe(value => {
```

Comment:
This subscription is never unsubsubscribed.

Reply:
It was the initial reproduction demo which I was going to clean after the first review but it stood like that. Cleaned it up.